tests/components/apache_kafka/test_init.py

```diff
@@ -0,0 +1,181 @@
+"""The tests for the Apache Kafka component."""
+from collections import namedtuple
+
+import pytest
+
+import homeassistant.components.apache_kafka as apache_kafka
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+
+from tests.async_mock import patch
+
+APACHE_KAFKA_PATH = "homeassistant.components.apache_kafka"
+PRODUCER_PATH = f"{APACHE_KAFKA_PATH}.AIOKafkaProducer"
+MIN_CONFIG = {
+    "ip_address": "localhost",
+    "port": 8080,
+    "topic": "topic",
+}
+FilterTest = namedtuple("FilterTest", "id should_pass")
+MockKafkaClient = namedtuple("MockKafkaClient", "init start send_and_wait")
+
+
+@pytest.fixture(name="mock_client")
+def mock_client_fixture():
+    """Mock the apache kafka client."""
+    with patch(f"{PRODUCER_PATH}.start") as start, patch(
+        f"{PRODUCER_PATH}.send_and_wait"
+    ) as send_and_wait, patch(f"{PRODUCER_PATH}.__init__", return_value=None) as init:
+        yield MockKafkaClient(init, start, send_and_wait)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def mock_client_stop():
+    """Mock client stop at module scope for teardown."""
+    with patch(f"{PRODUCER_PATH}.stop") as stop:
+        yield stop
+
+
+async def test_minimal_config(hass, mock_client):
+    """Test the minimal config and defaults of component."""
+    config = {apache_kafka.DOMAIN: MIN_CONFIG}
+    assert await async_setup_component(hass, apache_kafka.DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)